ember-core/src/main/scala/org/http4s/ember/core/EmberException.scala

```diff
@@ -39,4 +39,16 @@ object EmberException {
   final case class ChunkedEncodingError(message: String) extends EmberException {
     override def getMessage: String = message
   }
+
+  final case class EmptyStream() extends EmberException {
+    override def getMessage: String = "Cannot Parse Empty Stream"
+  }
+
+  final case class ReachedEndOfStream() extends EmberException {
```

Comment:
There's a standard `IOException` for this, but since `EmberException` is a `RuntimeException`, we can't extend it.